### PR TITLE
WSSQL-170 The antlr_target function will allow for rebuilding after m…

### PR DESCRIPTION
**Pull request by Michael-Gardner**
…odifying grammar

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

File: esp/services/ws_sql/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ################################################################################
-#    HPCC SYSTEMS software Copyright (C) 2014 HPCC Systems.
+#    HPCC SYSTEMS software Copyright(C)2014 HPCC Systems.
```
Comment (rpastrana) [inline esp/services/ws_sql/CMakeLists.txt]: Let's leave the copyright statements as-is

File: esp/services/ws_sql/CMakeLists.txt [hunk 2]
```diff
@@ -20,114 +20,100 @@
...
-PROJECT( ws_sql )
+project(ws_sql)
 
-# Java and ANTLR needed for generating lexer and parser
-FIND_PACKAGE(Java REQUIRED)
-FIND_PACKAGE(ANTLR REQUIRED)
+find_package(Java REQUIRED)
+find_package(ANTLR REQUIRED)
+include(${HPCC_WSSQL_SOURCE_DIR}/cmake_modules/buildANTLR3c.cmake)
+#include(${CMAKE_MODULE_PATH}FindANTLR3c.cmake)
```
Comment (rpastrana) [inline esp/services/ws_sql/CMakeLists.txt]: if we don't need this, let's remove it

File: cmake_modules/FindANTLR.cmake [hunk 3]
```diff
@@ -14,59 +14,112 @@
...
+    set(options "")
+    set(oneValueArgs GRAMMAR_PREFIX DESTINATION)
+    set(multiValueArgs GRAMMAR_FILES)
+    cmake_parse_arguments(antlr "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+    list(GET ARGN 0 target_name)
 
-    FILE(GLOB ANTLRRUNTIMECLASSES "${ANTLR_PATH}/ANTLR_PACKAGE_PATH/*.class")
+    if(antlr_DESTINATION)
+        # normalize path
+        string(REGEX REPLACE "[/]$" "" antlr_DESTINATION "${antlr_DESTINATION}")
```
Comment (Michael-Gardner) [inline cmake_modules/FindANTLR.cmake]: No it's useful for this function just to make sure the input is sane.

**Comment by Michael-Gardner** [comment]
@rpastrana Made the changes you recommended.

**Comment by rpastrana** [comment]
@Michael-Gardner you noted that this change needs documentation, did you open a doc issue to cover this?
Also, do these changes require any changes in the build environment and/or cmake command?

**Pull request merged**